discord/channel.py

```diff
@@ -324,7 +326,59 @@ def is_me(m):
 
                     count += 1
                     ret.append(msg)
+    @asyncio.coroutine
+    def webhooks(self):
+        """|coro|
+        
+        Gets all webhooks in this channel
+
+        Raises
+        -------
+        Forbidden
+            You do not have proper permissions to do the actions required.
+        HTTPException
+            Getting webhooks failed
+
 
+        Returns
+        --------
+        List
+            All webhooks in a channel, empty if none
+        """
+        webhooks = yield from self._state.http.get_channel_webhooks(self.id)
+        return [Webhook._from_data(self._state, wb) for wb in webhooks]
+    @asyncio.coroutine
+    def create_webhook(self,name,avatar):
+        """|coro|
+        Parameters
+        -----------
+        name: str
+            The name of the webhook
+        avatar: url or str
+            Avatar of the webhook. If str, it must be a base64 encoded image.
+
+        Raises
+        -------
+        Forbidden
+            You do not have proper permissions to do the actions required.
+        HTTPException
+            Creating the webhook failed.
+
+        Returns
+        --------
+        discord.Webhook - The created webhook
+        """
+        if isinstance(avatar,str):
+            ret = yield from self._state.http._session.get(avatar)
+            if ret.status not in [200,204]:
+                m = yield from ret.text()
+                raise discord.HTTPException(ret,message=m)
+            data = yield from ret.read()
+            img = io.BytesIO(data)
+            img.seek(0)
+            avatar = utils._bytes_to_base64_data(img.read())
```

Comment:
This entire segment probably belongs inside `HTTPClient`.

Or completely removed because nothing in the library allows setting a URL for an avatar/icon. Not to mention that this will probably always be invoked since the type hint says "URL or str" and you separate them as `isinstance(avatar, str)`, which will pretty much always be true.